db/migrations/00034_api_extensions.sql

```diff
@@ -0,0 +1,56 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE TYPE extension_status AS ENUM('online', 'offline');
+CREATE TYPE extension_resource_scope AS ENUM('user', 'system');
+
+CREATE TABLE extensions (
+  id UUID PRIMARY KEY NOT NULL DEFAULT gen_random_uuid(),
+  name STRING NOT NULL,
+  description STRING NOT NULL,
+  enabled BOOL NOT NULL,
+  slug STRING NOT NULL,
+	status extension_status NOT NULL DEFAULT 'offline',
+
+  created_at TIMESTAMPTZ NOT NULL DEFAULT now(),
+  updated_at TIMESTAMPTZ NOT NULL DEFAULT now(),
+  deleted_at TIMESTAMPTZ NULL,
+
+  CONSTRAINT extension_slug_key UNIQUE (slug) WHERE deleted_at IS NULL,
+  INDEX (slug, deleted_at) STORING (name, description, enabled, status, created_at, updated_at),
```

Comment:
Indices are suggested by cockroachDB when `explain` against the most common queries